core/backend/src/IModelDb.ts

```diff
@@ -707,16 +707,25 @@ export abstract class IModelDb extends IModel {
   }
 
   /**
-   * queries the BisCore.SubCategory table for the entries that are children of the passed categoryIds
+   * queries the BisCore.SubCategory table for the entries that are children of the passed categoryIds.
+   * If no iterable is passed, all subcategories that are part of parent categories containing an element are returned. If an empty iterable is passed, no subcategories are returned.
```

Comment:
> If no iterable is passed, all subcategories that are part of parent categories containing an element are returned.

This is confusing and inaccurate. You are only querying spatial categories and 3d elements.

You have changed the default behavior of this general-purpose function to serve one very specific purpose (ViewCreator3d).
Did you consider creating a new `queryAllUsedSpatialSubCategories` function instead?

Reply:
Thanks Paul, I wanted to initially explore an approach that would avoid creating a new rpc method, but I didn't pay attention to the guidelines for RPC versioning then. I've added a new `queryAllUsedSpatialSubCategories` instead.

I also originally thought to all parent categories were synonymous to all spatial categories and 3d elements because of an existing [method](https://github.com/iTwin/itwinjs-core/blob/4fc1a660752e5062c483b7280f440d5c380c418c/core/backend/src/CustomViewState3dCreator.ts#L42) we had in core backend. This is the same method used to get the parent categories for the subcategories of a default view. Maybe that method could be renamed to more accurately reflect its behavior